src/execution_plan/ops/op_delete.c

```diff
@@ -88,6 +88,9 @@ static OpResult DeleteInit(OpBase *opBase) {
 
 static Record DeleteConsume(OpBase *opBase) {
 	OpDelete *op = (OpDelete *)opBase;
+	if(op->op.childCount == 0) {
+		ErrorCtx_RaiseRuntimeException("Delete was called without a child operation");
+	}
```

Comment:
It would be better if we could perform this validation at the AST validation phase
adding this overhead to each `DeleteConsume` function is a bit of a shame